src/library/module.cpp

```diff
@@ -73,9 +85,13 @@ extern "C" LEAN_EXPORT object * lean_save_module_data(b_obj_arg fname, b_obj_arg
         // `MapViewOfFileEx` addresses must be aligned to the "memory allocation granularity", which is 64KB.
         base_addr = base_addr & ~((1LL<<16) - 1);
 
-        object_compactor compactor(reinterpret_cast<void *>(base_addr + strlen(g_olean_header) + sizeof(base_addr)));
+        object_compactor compactor(reinterpret_cast<void *>(base_addr + strlen(g_olean_header) + g_githash_header_size + sizeof(base_addr)));
         compactor(mdata);
         out.write(g_olean_header, strlen(g_olean_header));
+        out.write(LEAN_GITHASH, std::min(g_githash_header_size, strlen(LEAN_GITHASH)));
+        if (g_githash_header_size > strlen(LEAN_GITHASH)) {
+            out << std::string(g_githash_header_size - strlen(LEAN_GITHASH), '0');
```

Comment:
I would prefer if this was padded with null bytes or spaces, so that you can tell the difference between a missing githash and a real one, or hashes of different lengths.

Reply:
Also, why 32 bytes? Git hashes are 40 bytes, or 20 if you use a sensible encoding.